### PR TITLE
Your impact ipad

### PR DESCRIPTION
**Pull request by vauxhall**

### CODE REVIEW

**Review by ecotopian — COMMENTED** [review]
Great work! 👍  just one small concern regarding the margin calculation.

File: Client/Ecosia/UI/MyImpact/UICollectionView+maxWidth.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import UIKit
+
+extension UICollectionView {
+    var ecosiaHomeMaxWidth: CGFloat {
+        let insets = max(max(safeAreaInsets.left, safeAreaInsets.right), 16) * 2
+        let maxWidth = bounds.width - insets
+        
+        if UIDevice.current.userInterfaceIdiom == .pad {
```
[the remark this answers is not in the document]
Comment (vauxhall) [reply Client/Ecosia/UI/MyImpact/UICollectionView+maxWidth.swift:12]: Good reminder 👍

File: Client/Ecosia/UI/MyImpact/NewsController.swift [hunk 2]
```diff
@@ -133,8 +142,19 @@ final class NewsController: UIViewController, UICollectionViewDelegate, UICollec
     }
 }
 
+private extension UICollectionView {
+    func align(header: NewsSubHeader?) {
+        let margin = (bounds.width - ecosiaHomeMaxWidth) / 2
```
[the remark this answers is not in the document]
Comment (vauxhall) [reply Client/Ecosia/UI/MyImpact/NewsController.swift:147]: Instead of adding a cap every time we use this computed property I opted to have inside the implementation, so that it is safe to use, refer to [here](https://github.com/ecosia/ios-browser/blob/163037f843c4a3db573019e2a741027eaeec2d3a/Client/Ecosia/UI/MyImpact/UICollectionView%2BmaxWidth.swift#L10).

TL;DR `ecosiaHomeMaxWidth` can't be greater than `bounds.width`.

File: Client/Ecosia/UI/MyImpact/EcosiaHome.swift [hunk 3]
```diff
@@ -228,19 +207,35 @@ final class EcosiaHome: UICollectionViewController, UICollectionViewDelegateFlow
...
 
-    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
-        guard let section = Section(rawValue: section), section == .explore else { return .zero }
-        return .init(top: 26,
-                     left: max(collectionView.safeAreaInsets.left, 16),
-                     bottom: 26,
-                     right: max(collectionView.safeAreaInsets.right, 16))
+    func collectionView(_ collectionView: UICollectionView, layout: UICollectionViewLayout, insetForSectionAt: Int) -> UIEdgeInsets {
+        let vertical = insetForSectionAt == Section.explore.rawValue ? 26 : CGFloat()
+        let horizontal = (collectionView.bounds.width - collectionView.ecosiaHomeMaxWidth) / 2
```
[the remark this answers is not in the document]
Comment (vauxhall) [reply Client/Ecosia/UI/MyImpact/EcosiaHome.swift:212]: Same as before, better to have that logic on 1 place only, rather than 2 places.

**Review by ecotopian — APPROVED** [review]
